resources/karaf.rb

```diff
@@ -121,6 +111,13 @@ def client_command
     link '/etc/init.d/karaf' do
       to         "#{karaf_path}/bin/karaf-service"
       link_type  :symbolic
+	  only_if    { ::File.exists?("#{karaf_path}/bin/karaf-service") }
+    end
+	
+	link '/etc/init.d/karaf' do
```

Comment:
Looking at this again, this is actually overwriting the previous symbolic link at /etc/init.d/karaf. At least for Ubuntu 14 (not sure about later), it needs the karaf-service file as the link.